From: Huang Ying <ying.huang@intel.com>
Subject: mm/memory.c: improve readability of clear_huge_page

The optimized clear_huge_page() isn't easy to read and understand.  This
is suggested by Michael Hocko to improve it.

Link: http://lkml.kernel.org/r/20170830051842.1397-1-ying.huang@intel.com
Signed-off-by: "Huang, Ying" <ying.huang@intel.com>
Suggested-by: Michal Hocko <mhocko@suse.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/memory.c |   11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff -puN mm/memory.c~mm-improve-readability-of-clear_huge_page mm/memory.c
--- a/mm/memory.c~mm-improve-readability-of-clear_huge_page
+++ a/mm/memory.c
@@ -4543,12 +4543,15 @@ void clear_huge_page(struct page *page,
 	 * towards the sub-page to access
 	 */
 	for (i = 0; i < l; i++) {
+		int left_idx = base + i;
+		int right_idx = base + 2 * l - 1 - i;
+
 		cond_resched();
-		clear_user_highpage(page + base + i,
-				    addr + (base + i) * PAGE_SIZE);
+		clear_user_highpage(page + left_idx,
+				    addr + left_idx * PAGE_SIZE);
 		cond_resched();
-		clear_user_highpage(page + base + 2 * l - 1 - i,
-				    addr + (base + 2 * l - 1 - i) * PAGE_SIZE);
+		clear_user_highpage(page + right_idx,
+				    addr + right_idx * PAGE_SIZE);
 	}
 }
 
_
